driver-core/src/main/com/mongodb/client/model/expressions/ArrayExpression.java

```diff
@@ -58,67 +57,267 @@
      */
     IntegerExpression size();
 
+    /**
+     * True if any value in {@code this} array satisfies the predicate.
+     *
+     * @param predicate the predicate.
+     * @return the resulting value.
+     */
     BooleanExpression any(Function<? super T, BooleanExpression> predicate);
 
+    /**
+     * True if all values in {@code this} array satisfy the predicate.
+     *
+     * @param predicate the predicate.
+     * @return the resulting value.
+     */
     BooleanExpression all(Function<? super T, BooleanExpression> predicate);
 
+    /**
+     * The sum of adding together all the values of {@code this} array,
+     * via the provided {@code mapper}. Returns 0 if the array is empty.
+     *
+     * <p>The mapper may be used to transform the values of {@code this} array
+     * into {@linkplain NumberExpression numbers}. If no transformation is
+     * necessary, then the identify function {@code array.sum(v -> v)} should
+     * be used.
+     *
+     * @param mapper the mapper function.
+     * @return the resulting value.
+     */
     NumberExpression sum(Function<? super T, ? extends NumberExpression> mapper);
```

Comment:
[here and in other places]

1. identify -> identity
2. [optional] `the identify function {@code array.sum(v -> v)} should be used` -> `the {@linkplain Function#identity() identity function} should be used`?

Reply:
Good catch. (For 2, I will omit - I think the code example is more helpful, and the v->v format more legible)